pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -364,6 +365,78 @@ func buildFakeServer(t *testing.T) string {
 	return filepath.Dir(serverBin)
 }
 
+const (
+	pythonScript = `#! /usr/bin/python
+
+import socket
+import time
+
+HOST = '127.0.0.1'
+PORT = # Empty port
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.bind((HOST, PORT))
+s.listen()
+time.sleep(30)
+`
+)
+
+func TestPythonFromBashScript(t *testing.T) {
+	t.Run("PythonFromBashScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, "python-*.sh", []byte(pythonScript), []string{"sh", "-c"}, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+	t.Run("DirectPythonScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, "script.py", []byte(pythonScript), nil, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+}
+
+func testCaptureWrappedCommands(t *testing.T, scriptName string, scriptContent []byte, commandWrapper []string, validator func(service model.Service) bool) {
+	// Creating tempfile
+	tempfile, err := os.CreateTemp("", scriptName)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		tempfile.Close()
+		os.Remove(tempfile.Name())
+	})
+	_, err = tempfile.Write(scriptContent)
+	require.NoError(t, err)
+	require.NoError(t, tempfile.Close())
+
+	// Changing permissions
+	require.NoError(t, os.Chmod(tempfile.Name(), 0755))
+
+	commandLine := append(commandWrapper, tempfile.Name())
+	cmd := exec.Command(commandLine[0], commandLine[1:]...)
+	// Running the binary in the background
+	require.NoError(t, cmd.Start())
+
+	var proc *process.Process
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		proc, err = process.NewProcess(int32(cmd.Process.Pid))
+		assert.NoError(collect, err)
+	}, 10*time.Second, 100*time.Millisecond)
+
+	if len(commandWrapper) > 0 {
+		children, err := proc.Children()
+		require.NoError(t, err)
+		require.Len(t, children, 1)
```

Comment:
Shouldn't this be inside the eventually loop above too to avoid races?

Reply:
done